src/main/java/edu/kis/powp/jobs2d/drivers/visitor/DriverDetailsVisitor.java

```diff
@@ -27,6 +29,12 @@ public void visit(DriversComposite driverComposite) {
         String details = "\nDriversComposite details:" +
                 "\n Registered drivers count: " + registeredDrivers;
         stringBuilder.append(details);
+
+        Iterator<VisitableJob2dDriver> driverIterator = driverComposite.getIterator();
+        while(driverIterator.hasNext()) {
+            VisitableJob2dDriver nextDriver = driverIterator.next();
+            nextDriver.accept(this);
+        }
```

Comment:
OK. However, implementation of visit methods does not follow a stateless approach. Fortunately, getDetails method makes usage workflow correct.